src/platform/plugins/shared/workflows_management/public/features/run_workflow/ui/workflow_execute_manual_form.tsx

```diff
@@ -0,0 +1,172 @@
+  return z.object(
+    inputs.reduce((acc, input) => {
+      switch (input.type) {
+        case 'string':
+          acc[input.name] = input.required ? z.string() : z.string().optional();
+          break;
+        case 'number':
+          acc[input.name] = input.required ? z.number() : z.number().optional();
+          break;
+        case 'boolean':
+          acc[input.name] = input.required ? z.boolean() : z.boolean().optional();
+          break;
+        case 'choice':
+          acc[input.name] = input.required
+            ? z.enum(input.options as [string, ...string[]])
+            : z.enum(input.options as [string, ...string[]]).optional();
+          break;
+      }
+      return acc;
+    }, {} as Record<string, z.ZodType>)
+  );
+};
+
+interface WorkflowExecuteManualFormProps {
+  workflow: WorkflowDetailDto | WorkflowListItemDto;
+  value: string;
+  setValue: (data: string) => void;
+  errors: string | null;
+  setErrors: (errors: string | null) => void;
+}
+
+const defaultWorkflowInputsMappings: Record<string, any> = {
+  string: 'Enter a string',
+  number: 0,
+  boolean: false,
+  choice: (input: any) => `Select an option: ${input.options.join(', ')}`,
+};
+
+const getDefaultWorkflowInput = (workflow: WorkflowDetailDto | WorkflowListItemDto): string => {
+  const inputPlaceholder: Record<string, any> = {};
+
+  if (workflow?.definition.inputs) {
+    workflow.definition.inputs.forEach((input: any) => {
+      let placeholder: string | number | boolean | ((input: any) => string) =
+        defaultWorkflowInputsMappings[input.type];
+      if (typeof placeholder === 'function') {
+        placeholder = placeholder(input);
+      }
+      inputPlaceholder[input.name] = input.default || placeholder;
+    });
+  }
+
+  return JSON.stringify(inputPlaceholder, null, 2);
+};
+
+export const WorkflowExecuteManualForm = ({
+  workflow,
+  value,
+  setValue,
+  errors,
+  setErrors,
+}: WorkflowExecuteManualFormProps): React.JSX.Element => {
+  const inputsValidator = useMemo(
+    () => makeWorkflowInputsValidator(workflow.definition.inputs || []),
+    [workflow?.definition.inputs]
+  );
+
+  const handleChange = useCallback(
+    (data: string) => {
+      setValue(data);
+      if (workflow?.definition.inputs) {
+        try {
+          const res = inputsValidator.safeParse(JSON.parse(data));
+          if (!res.success) {
+            setErrors(
+              res.error.issues.map((e: any) => `${e.path.join('.')}: ${e.message}`).join(', ')
+            );
+          } else {
+            setErrors(null);
+          }
+        } catch (e: Error | any) {
+          setErrors(`Invalid JSON: ${e.message || e.toString()}`);
+        }
+      }
+    },
+    [setValue, workflow?.definition.inputs, inputsValidator, setErrors]
+  );
+
+  useEffect(() => {
+    if (!value && workflow) {
+      handleChange(getDefaultWorkflowInput(workflow));
+    }
+  }, [workflow, value, handleChange]);
+
+  return (
```

Comment:
if there's no inputs defined in workflow I think we should just show subdued message "No inputs required for this workflow" and don't show the json editor

https://github.com/user-attachments/assets/ebfd7439-73b1-4ebc-a406-30ec0a94760d